#!/usr/bin/env python
PACKAGE = "two_d_guidance"
    
from dynamic_reconfigure.parameter_generator_catkin import *
    
gen = ParameterGenerator()
    
mode_enum = gen.enum([ gen.const("Staging",   int_t, 0, "Staging"),
                       gen.const("Ready",     int_t, 1, "Ready"),
                       gen.const("Racing",    int_t, 2, "Racing"),
                       gen.const("Finished",  int_t, 3, "Finished")],
                     "An enum to set race mode")
   
gen.add("mode", int_t, 0, "Race mode", 0, 0, 4, edit_method=mode_enum)

gen.add("nb_lap",  int_t, 0, "Total number of laps", 2, 1, 5)
gen.add("cur_lap", int_t, 0, "Total number of laps", 0, 0, 5)


exit(gen.generate(PACKAGE, "two_d_guidance", "trr_race_manager"))
